### PR TITLE
(RE-13191) Use new directory throughout the msi signing

### PR DESCRIPTION
**Pull request by underscorgan**
Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

File: lib/packaging/sign/msi.rb [hunk 1]
```diff
@@ -65,7 +65,8 @@ def sign(target_dir = 'pkg')
     # first Sha1 signature.
     sign_command = <<-CMD
 for msi in #{msis.map { |d| File.dirname(d) + "/" + File.basename(d) }.join(" ")}; do
-  if "/cygdrive/c/tools/osslsigncode-fork/osslsigncode.exe" verify -in "C:/#{work_dir}/$msi" ; then
+  msidir="C:/#{work_dir}/$(dirname $msi)"
+  if "/cygdrive/c/tools/osslsigncode-fork/osslsigncode.exe" verify -in "$msidir/$msi" ; then
```
Comment (shrug) [inline lib/packaging/sign/msi.rb]: does $msi already contain the $msidir directory?

File: lib/packaging/sign/msi.rb [hunk 2]
```diff
@@ -8,11 +8,11 @@ def sign(target_dir = 'pkg')
...
     work_dir = "Windows/Temp/#{Pkg::Util.rand_string}"
+    Pkg::Util::Net.remote_ssh_cmd(ssh_host_string, "mkdir -p C:/#{work_dir}")
     msis = Dir.glob("#{target_dir}/windows*/**/*.msi")
-    msis.each do | msi |
-      Pkg::Util::Net.remote_ssh_cmd(ssh_host_string, "mkdir -p C:/#{work_dir}/#{File.dirname(msi)}")
-      Pkg::Util::Net.rsync_to(msi, rsync_host_string, "/cygdrive/c/#{work_dir}/#{File.dirname(msi)}")
-    end
+    Pkg::Util::Net.rsync_to(msis.join(" "), rsync_host_string, "/cygdrive/c/#{work_dir}",
+                           extra_flags: ["--ignore-existing --relative"])
+
```
Comment (ciprianbadescu) [inline lib/packaging/sign/msi.rb]: I reverted this to before https://github.com/puppetlabs/packaging/pull/1028 rsync command with extra-flag='--relative' that will keep the directory structure (--ignore-existing was default).

File: lib/packaging/sign/msi.rb [hunk 3]
```diff
@@ -64,8 +64,10 @@ def sign(target_dir = 'pkg')
     # Once we no longer support Windows 8/Windows Vista, we can remove the
     # first Sha1 signature.
     sign_command = <<-CMD
-for msi in #{msis.map { |d| File.dirname(d) + "/" + File.basename(d) }.join(" ")}; do
```
Comment (ciprianbadescu) [inline lib/packaging/sign/msi.rb]: `File.dirname(d) + "/" + File.basename(d)` => funny code I wrote last evening ...

**Comment by underscorgan** [comment]
Was able to successfully test this out and confirm it is working \o/

Will push changelog update then we should be able to get this in and released.

**Pull request merged**